code/game/objects/items/nemesis_items.dm

```diff
@@ -0,0 +1,458 @@
+
+	active = TRUE
+	icon_state = "trap_deployed"
+	dir = get_dir(target, user)
+	user.dropItemToGround(src, TRUE)
+	update_icon()
+	anchored = TRUE
+	playsound(src, 'sound/magic/blind.ogg', 50, TRUE)
+	deploy_beams()
+
+/obj/item/nemesis_trap/attack_hand(mob/living/user, list/modifiers)
+	if(!active)
+		return ..()
+
+	to_chat(user, "<span class='warning'>You attempt to disarm [src]...</span>")
+	var/disarm_time = 15 SECONDS
+	if(ishuman(user))
+		var/mob/living/carbon/human/disarmer = user
+		if(istype(disarmer.get_item_by_slot(ITEM_SLOT_GLOVES), /obj/item/clothing/gloves/rapid/nemesis))
+			disarm_time = 1 SECONDS
+
+	if(!do_after(user, disarm_time, src))
+		return
+
+	to_chat(user, "<span class='notice'>You successfully disarm [src].</span>")
+	qdel(src)
+
+/obj/item/nemesis_trap/attackby(obj/O, mob/user, params)
+	if(!active)
+		return ..()
+
+	if(!istype(O, /obj/item/screwdriver))
+		return ..()
+
+	to_chat(user, "<span class='warning'>You attempt to disarm [src]...</span>")
+	var/disarm_time = 3 SECONDS
+	if(ishuman(user))
+		var/mob/living/carbon/human/disarmer = user
+		if(istype(disarmer.get_item_by_slot(ITEM_SLOT_GLOVES), /obj/item/clothing/gloves/rapid/nemesis))
+			disarm_time = 1 SECONDS
+
+	if(!do_after(user, disarm_time, src))
+		return
+
+	to_chat(user, "<span class='notice'>You successfully disarm [src].</span>")
+	qdel(src)
+
+/obj/item/nemesis_trap/proc/deploy_beams()
+	for(var/obj/item/nemesis_trap/trap in view(3, get_turf(src)))
+		if(trap == src || !trap.active)
+			continue
+
+		icon_state = "trap_deployed_active"
+		update_icon()
+
+		trap.icon_state = "trap_deployed_active"
+		trap.update_icon()
+
+		var/datum/beam/beam = Beam(trap, "nemesis", maxdistance = 5, beam_type = /obj/effect/ebeam/nemesis)
+		beams += beam
+		trap.beams += beam
+
+/obj/item/nemesis_trap/proc/trigger()
+	var/datum/effect_system/spark_spread/sparks = new(get_turf(src))
+	sparks.set_up(4, 0, loc)
+	sparks.start()
+	qdel(src)
+
+/obj/item/nemesis_trap/Destroy()
+	for(var/datum/beam/beam in beams)
+		QDEL_NULL(beam)
+	. = ..()
+
+/obj/effect/ebeam/nemesis
+	name = "energy beam"
+	desc = "An energy beam, connected to a disk trap."
+
+	mouse_opacity = MOUSE_OPACITY_ICON
+
+/obj/effect/ebeam/nemesis/Crossed(atom/movable/AM)
+	. = ..()
+	if(isliving(AM))
+		var/mob/living/target = AM
+		if(target.body_position == LYING_DOWN)
+			return
+
+		var/turf/cur_loc = get_turf(src)
+		var/obj/item/nemesis_trap/trap_one = owner.origin
+		var/obj/item/nemesis_trap/trap_two = owner.target
+
+		new /obj/effect/temp_visual/nemesis_circle/huge(cur_loc)
+
+		to_chat(target, "<span class='userdanger'>You cross [src] and it destabilises, releasing a large energy blast!</span>")
+		playsound(cur_loc, 'sound/magic/disable_tech.ogg', 50, TRUE)
+
+		for(var/mob/living/victim in range(1, cur_loc))
+			if(victim != target)
+				to_chat(victim, "<span class='userdanger'>You are hit by an energy blast!</span>")
+			victim.apply_damage(35, STAMINA)
```

Comment:
wow, this is dogshit!

like, I could maybe consider it if the beam nodes weren't deleted afterwards, but unless you stack like 3 of these on one tile and your enemy is dumb enough to walk through a very obvious beam, you're permanently sacrificing two nodes for some temporary stamina damage that will wear off in 10 seconds (unless you're there to follow up on it). said stamina damage isn't even enough to cause damage slowdown, nor is it enough to make any of your other stamina weaponry (except your non-overloaded mines (which are better off NOT having their positions given away by a giant beam of light) and your stamina punches (which you shouldn't be using against most targets anyway, since the disarms are so much beter)) take fewer shots/uses to take someone down, as far as I'm aware.

Reply:
Fixed that. Now each trap can only have one connection and they can't be stacked